### PR TITLE
Rule Mapping - Create and Assign

### PR DESCRIPTION
**Pull request by pooja-koul-incontact**

### CODE REVIEW

File: src/app/bu/bu-rule-mapping/bu-create-rule-modal/bu-create-rule-modal.component.css [hunk 1]
```diff
@@ -0,0 +1,388 @@
+.modal-dialog
```
Comment (KMayuri25) [inline src/app/bu/bu-rule-mapping/bu-create-rule-modal/bu-create-rule-modal.component.css]: container name

File: src/app/bu/bu-rule-mapping/bu-create-rule-modal/bu-create-rule-modal.component.ts [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+    $('.dd3 li a').click(function () {
+      $('#externalStateDropdownMenu .selected-text').html($(this).text());
+      $('#externalStateDropdownMenu .selected-text').prop('title', $(this).text());
+      $('.dd3 li a').removeClass('active');
+      $(this).addClass('active');
+    });
+    this.createRuleForm = this.formBuilder.group({
+      ruleName: ['', Validators.required]
+    });
+    this.createNewRuleService.getUnassignedUsers().subscribe(
```
Comment (Susmith-roy-incontact) [inline src/app/bu/bu-rule-mapping/bu-create-rule-modal/bu-create-rule-modal.component.ts]: Can we call this service when we really need this,

File: src/app/bu/bu-rule-mapping/bu-rule-mapping-rule-name/bu-rule-mapping-rule-name.component.spec.ts [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+  beforeEach(async(() => {
+    TestBed.configureTestingModule({
+      imports: [RouterTestingModule],
+      declarations: [ BuRuleMappingRuleNameComponent ]
+    })
+    .compileComponents();
+  }));
+
+  beforeEach(() => {
```
Comment (KMayuri25) [inline src/app/bu/bu-rule-mapping/bu-rule-mapping-rule-name/bu-rule-mapping-rule-name.component.spec.ts]: use async

File: src/app/bu/bu-rule-mapping/bu-rule-mapping-type/bu-rule-mapping-type.component.ts [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    createdTime: '',
+    userAssigned: ''
+  };
+  ngOnInit() {
+  }
+  agInit(params) {
+    this.rowData = params.data;
+  }
+  onRuleTypeClick() {
+    localStorage.setItem('updateRuleMapping', JSON.stringify(this.rowData));
```
Comment (KMayuri25) [inline src/app/bu/bu-rule-mapping/bu-rule-mapping-type/bu-rule-mapping-type.component.ts]: remove when not required

File: src/app/bu/bu-rule-mapping/bu-rule-mapping-creation-date/bu-rule-mapping-creation-date.component.ts [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+    createdTime: '',
+    userAssigned: ''
+  };
+  ngOnInit() {
+  }
+  agInit(params) {
+    this.rowData = params.data;
+  }
+  onRuleCreationDateClick() {
+    localStorage.setItem('updateRuleMapping', JSON.stringify(this.rowData));
```
Comment (Susmith-roy-incontact) [inline src/app/bu/bu-rule-mapping/bu-rule-mapping-creation-date/bu-rule-mapping-creation-date.component.ts]: remove this

File: src/app/bu/bu-rule-mapping/bu-rule-mapping-rule-name/bu-rule-mapping-rule-name.component.ts [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+    createdTime: '',
+    userAssigned: ''
+  };
+  ngOnInit() {
+  }
+  agInit(params) {
+    this.rowData = params.data;
+  }
+    onRuleNameClick() {
+    localStorage.setItem('updateRuleMapping', JSON.stringify(this.rowData));
```
Comment (Susmith-roy-incontact) [inline src/app/bu/bu-rule-mapping/bu-rule-mapping-rule-name/bu-rule-mapping-rule-name.component.ts]: remove this when done

File: src/app/bu/bu-rule-mapping/bu-rule-mapping-status/bu-rule-mapping-status.component.ts [hunk 7]
```diff
@@ -27,13 +27,17 @@ export class BuRuleMappingStatusComponent implements OnInit {
...
+    if (!word) {
+      return word;
     }  else {
-      return status[0].toUpperCase() + status.substr(1).toLowerCase();
+      return word[0].toUpperCase() + word.substr(1).toLowerCase();
     }
   }
-
+  onRuleStatusClick() {
+    localStorage.setItem('updateRuleMapping', JSON.stringify(this.rowData));
```
Comment (Susmith-roy-incontact) [inline src/app/bu/bu-rule-mapping/bu-rule-mapping-status/bu-rule-mapping-status.component.ts]: Remove when done

**Pull request merged**